### PR TITLE
remove driver object from node dictionary

### LINKED ISSUE #38195: TypeError: can't serialize <libcloud.compute.drivers.gce.GCENodeDriver object at 0x3b00390>
### Description of Issue/Question
I'm trying to deploy vm in gce via cloud states and I face with error
```TypeError: can't serialize <libcloud.compute.drivers.gce.GCENodeDriver object at 0x3b00390>```
### Setup
Install master and salt-cloud
cloud.sls
```
cloud:
  ssh_key_name: saltstack
  ssh_key_file: /root/.ssh/id_rsa
  update_cachedir: True
  diff_cache_events: True
  change_password: True

  providers:
    paas-gce-acme:
      driver: gce
       some gce options
```

paas.sls
```
testing:
  cloud:
    - present
    - cloud_provider:  paas-gce-acme
    - image: centos-7
    - size: f1-micro
    - location: europe-west1-b
    - network: default
    - use_persistent_disk: True
```

### Steps to Reproduce Issue
salt-run state.sls paas  -l debug
```
[DEBUG   ] Rendered data from file: /var/cache/salt/master/files/base/paas/init.sls:


testing:
  cloud:
    - present
    - cloud_provider:  paas-gce-acme
    - image: centos-7
    - size: f1-micro
    - location: europe-west1-b
    - network: default
    - use_persistent_disk: True

[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/state.py", line 1744, in call
    **cdata['kwargs'])
  File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1702, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/states/cloud.py", line 117, in present
    if __salt__['cloud.has_instance'](name=name, provider=None):
  File "/usr/lib/python2.7/site-packages/salt/modules/cloud.py", line 150, in has_instance
    data = get_instance(name, provider)
  File "/usr/lib/python2.7/site-packages/salt/modules/cloud.py", line 176, in get_instance
    data = action(fun='show_instance', names=[name], provider=provider)
  File "/usr/lib/python2.7/site-packages/salt/modules/cloud.py", line 237, in action
    info = client.action(fun, cloudmap, names, provider, instance, kwargs)
  File "/usr/lib/python2.7/site-packages/salt/cloud/__init__.py", line 485, in action
    return mapper.do_action(names, kwargs)
  File "/usr/lib/python2.7/site-packages/salt/cloud/__init__.py", line 1532, in do_action
    vm_name, call='action'
  File "/usr/lib/python2.7/site-packages/salt/cloud/clouds/gce.py", line 284, in show_instance
    __utils__['cloud.cache_node'](node, __active_provider_name__, __opts__)
  File "/usr/lib/python2.7/site-packages/salt/utils/cloud.py", line 2902, in cache_node
    msgpack.dump(node, fh_)
  File "/usr/lib64/python2.7/site-packages/msgpack/__init__.py", line 38, in pack
    stream.write(packer.pack(o))
  File "msgpack/_packer.pyx", line 231, in msgpack._packer.Packer.pack (msgpack/_packer.cpp:3661)
  File "msgpack/_packer.pyx", line 233, in msgpack._packer.Packer.pack (msgpack/_packer.cpp:3503)
  File "msgpack/_packer.pyx", line 192, in msgpack._packer.Packer._pack (msgpack/_packer.cpp:2657)
  File "msgpack/_packer.pyx", line 192, in msgpack._packer.Packer._pack (msgpack/_packer.cpp:2657)
  File "msgpack/_packer.pyx", line 192, in msgpack._packer.Packer._pack (msgpack/_packer.cpp:2657)
  File "msgpack/_packer.pyx", line 228, in msgpack._packer.Packer._pack (msgpack/_packer.cpp:3382)
TypeError: can't serialize <libcloud.compute.drivers.gce.GCENodeDriver object at 0x3b00390>

[INFO    ] Completed state [testing] at time 18:22:50.739519 duration_in_ms=45210.669
[DEBUG   ] File /var/cache/salt/master/accumulator/61189008 does not exist, no need to cleanup.
[DEBUG   ] LazyLoaded config.option
```
### Versions Report
```
salt-run --versions-report
Salt Version:
           Salt: 2016.11.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Sep 15 2016, 22:37:39)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.36.3.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core
 
[root@localhost ~]# salt --versions-report
Salt Version:
           Salt: 2016.11.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Sep 15 2016, 22:37:39)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.36.3.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core

```
```
pip freeze
apache-libcloud==1.4.0
Babel==0.9.6
backports.ssl-match-hostname==3.4.0.2
chardet==2.2.1
configobj==4.7.2
decorator==3.4.0
futures==3.0.5
iniparse==0.4
Jinja2==2.7.2
kitchen==1.1.1
MarkupSafe==0.11
msgpack-python==0.4.8
perf==0.1
pycrypto==2.6.1
pycurl==7.19.0
pygobject==3.14.0
pygpgme==0.3
pyliblzma==0.5.3
pyudev==0.15
pyxattr==0.5.1
PyYAML==3.11
pyzmq==15.3.0
requests==2.6.0
salt==2016.11.0
six==1.9.0
slip==0.4.0
slip.dbus==0.4.0
tornado==4.2.1
urlgrabber==3.10
urllib3==1.10.2
yum-metadata-parser==1.1.4
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?

Make sure the node object passed to the cloud cache is serializable.

### What issues does this PR fix or reference?

Fixes #38195

### CODE REVIEW

**Comment by gtmanfred** [comment]
@erjohnso Do you reuse the `driver` object on the node later in the libcloud driver?

Thanks,
Daniel

**Comment by erjohnso** [comment]
It's been quite some time since my last foray into salt, but I think these nested driver objects is an artifact of libcloud. I could be mistaken, but iirc, most of the classes have a driver instance variable.

Example: [GCEAddress](https://github.com/apache/libcloud/blob/trunk/libcloud/compute/drivers/gce.py#L302)

is there something new in Salt with how it does caching / marshalling that surfaced this bug?

**Comment by gtmanfred** [comment]
It just looks like the gce driver was never fully tested with the cloud node cache stuff when it was being added.  Because this would have never worked.

It is odd, because we don't see the same issue with the openstack driver which would use libcloud as well.
https://docs.saltstack.com/en/latest/topics/cloud/misc.html#salt-cloud-cache

I will double check in the upstream libcloud and see if we even need the driver left on the node, or not and if not then I will remove the driver key/value pairs in the expand_node function.

File: salt/cloud/clouds/gce.py [hunk 1]
```diff
@@ -47,6 +47,7 @@
 
 # Import python libs
 from __future__ import absolute_import
+import copy
```
Comment (mirceaulinic) [inline salt/cloud/clouds/gce.py]: Where is this used?

**Comment by gtmanfred** [comment]
It was used before, it can be removed.

I am going to rewrite it once I get a chance to look into libcloud gce
driver more.

**Pull request closed without merging**